pkg/ipcache/ipcache.go

```diff
@@ -225,6 +225,12 @@ func (ipc *IPCache) GetHostIP(ip string) net.IP {
 	return hostIP
 }
 
+func (ipc *IPCache) GetHostIPCache(ip string) (net.IP, uint8) {
+	ipc.mutex.RLock()
+	defer ipc.mutex.RUnlock()
```

Comment:
This doesn't guarantee that the caller is able to obtain the lock, so why perform this lock/unlock here?

Reply:
Ah nevermind, based on the name of the functions it looked like this was returning a handle to the entire cache. This is definitely nitpicky, but I would call these `GetHostCachedIP` and `getHostCachedIP`